dts/espressif/32/pinctrl.yaml

```diff
@@ -0,0 +1,101 @@
+# According to ESP32's TRM:
+# Note that the I/O GPIO pads are 0-19, 21-23, 25-27, 32-39, 
+# while the output GPIOs are 0-19, 21-23, 25-27, 32-33.
+# GPIO pads 34-39 are input­only.
+
+uart0:
+  tx:
+    func: [out]
+    gpio: [[0, 19], [21, 23], [25, 27], [32, 33]]
+    sigo: u0txd_out
+  rx:
+    func: [in]
+    bias: pull_up
```

Comment:
Please, check which pins for PU/PD pins. AFAIK, ESP32 pins 34~39 have no PU/PD pins. This should be checked for all SoCs.